apis/python/src/tiledbsoma/_query.py

```diff
@@ -812,7 +813,10 @@ def _read_as_csr(
 
     d0_joinids = d0_joinids_arr.to_numpy()
     d1_joinids = d1_joinids_arr.to_numpy()
-    nnz = matrix.nnz
+    try:
+        nnz = matrix._handle._handle.nnz(raise_if_slow=True)
+    except SOMAError:
+        nnz = -1
```

Comment:
`None` would be more Pythonic and less likely to get accidentally used in arithmetic by some future maintainer of this code

Reply:
I was doing this to also handle the case where nnz == 0.  I'm not sure None is any more pythonic - could you cite a source that applies to this case?

Reply:
to clarify, we need to handle:
* nnz == 0, empty array
* nnz > 0
* no nnz avaialble
making it polymorphic means that tests below go from `nnz > 0` to `nnz is not None and nnz > 0`, which seems cumbersome

Reply:
OK

Reply:
OK, will do.  I register my disagreement, but I think getting this landed has priority over that.